### DISCUSSION TITLE
Command to create new note?

### DISCUSSION THREAD

**Discussion by ddbrierton** [nvim-neorg/neorg#954] [Q&A] [answered]
I wanted to create a keymap for creating a new note but I can’t figure out what the command is. I know that it is bound to `<leader> n n n` but what is the actual command? Looking at the code that seems to be bound to `core.dirman.new.note` but that doesn’t seem to be available outside of Neorg. I’m a little surprised that we have `:Neorg index` and `:Neorg journal` but not `:Neorg new` — wouldn’t that be a good idea?

**Comment by benlubas** [comment]
idk what the command is either, but there's a remap key function that's part of the [keybind](https://github.com/nvim-neorg/neorg/wiki/User-Keybinds#setting-up-a-keybind-hook) core module.

```lua
-- Want to move one keybind into the other? `remap_key` moves the data of the
-- first keybind to the second keybind, then unbinds the first keybind.
keybinds.remap_key("norg", "n", "<C-Space>", "<Leader>t")
```

The default bind should be `<localleader>nn` so something like this should work:
```lua
keybinds.remap_key("norg", "n", "<localleader>nn", "whatever you want")
```

**Reply by ddbrierton** [reply]
Thank you, that’s a genuinely useful tip!

Unfortunately, what I was hoping to figure out was if there was a way to create a new note when not in Neorg mode — the sort of use case where you're working on something and think, “oh, I’d better make a note about this.” At the moment I have to first open my index file so that I’m in Neorg mode, and then I can create a new note.

Perhaps this just isn’t possible right now — I understand that they are working on fundamentals at the moment, but I would’ve thought that creating a new note was pretty basic functionality.

**Reply by benlubas** [reply]
I think there is no good way to do this. Like no intended way? But there is a way.

This will create and open a file in the current neorg workspace. You can give a full path. It adds the norg extension for you.
```lua
require('neorg.modules.core.dirman.module').public.create_file('path/to/file/name') 
```
Useful if you would like to generate a note with a name that's relative to the file that you're in or something like that.

If you want to trigger the popup, you would have to define your own function to do that. I took this [source code](https://github.com/nvim-neorg/neorg/blob/9a55117b6175e63d86346bb893392cdf0b6ad2cc/lua/neorg/modules/core/dirman/module.lua#L512-L523) and modified it slightly to make this function which works for me

```lua
local function new_note()
  local ui = require("neorg.modules.core.ui.module")
  local dirman = require("neorg.modules.core.dirman.module")
  ui.public.create_prompt("NeorgNewNote", "New Note: ", function(text)
    dirman.public.create_file(text)
  end, {
    center_x = true,
    center_y = true,
  }, {
    width = 25,
    height = 1,
    row = 10,
    col = 0,
  })
end
```

**Reply by benlubas** [reply, marked as answer]
Wait nevermind. you can just map `:Neorg keybind norg core.dirman.new.note<CR>`

**Reply by ddbrierton** [reply]
Damn that was some good detective work! Thank you, that does indeed work perfectly.
